### PR TITLE
fixed ability to delete slot on availability page

### PR DESCRIPTION
**Pull request by Pavliuk13**
Slots are reloaded when user deletes slot

### CODE REVIEW

File: frontend/src/app/modules/availability/user-slot/user-slot.component.ts [hunk 1]
```diff
@@ -13,6 +15,26 @@ export class UserSlotComponent {
 
     @Input() public currentUser: IUser;
 
+    protected destroyed$ = new Subject<void>();
+
+    readonly untilThis = <T>(source: Observable<T>) => source.pipe(takeUntil(this.destroyed$));
+
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/user-slot/user-slot.component.ts]: Why did you add this?

File: frontend/src/app/modules/availability/user-slot/user-slot.component.ts [hunk 2]
```diff
@@ -13,6 +15,26 @@ export class UserSlotComponent {
...
+                this.userSlots = resp.userSlots;
+                this.spinnerService.hide();
+            });
+    }
+
+    isDeleted(isRemove: any) {
+        if (isRemove) {
+            this.updateUserSlots();
+        }
+    }
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/user-slot/user-slot.component.ts]: I believe we need the same logic for team slots

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/availability/team/team.component.ts [hunk 3]
```diff
@@ -11,15 +14,37 @@ import { SpinnerService } from '@core/services/spinner.service';
 export class TeamComponent implements OnInit {
     @Input() public teamSlot: IAvailabilitySlotsGroupByTeamsDto;
 
+    @Input() public currentUser: IUser;
+
     public slots: Array<IAvailabilitySlot>;
 
+    private unsubscribe$ = new Subject<void>();
+
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/team/team.component.ts]: I believe you don't need this

File: frontend/src/app/modules/availability/team/team.component.ts [hunk 4]
```diff
@@ -11,15 +14,37 @@ import { SpinnerService } from '@core/services/spinner.service';
...
+    constructor(public spinnerService: SpinnerService, private availabilitySlotService: AvailabilitySlotService) {}
 
     ngOnInit(): void {
         this.slots = this.teamSlot.availabilitySlots;
         this.teamName = this.teamSlot.name;
     }
+
+    public updateTeamsSlots() {
+        this.availabilitySlotService
+            .getUserPersonalAndTeamSlots(this.currentUser.id)
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/team/team.component.ts]: We have a separate card for this

File: frontend/src/app/modules/availability/team/team.component.ts [hunk 5]
```diff
@@ -11,15 +14,37 @@ import { SpinnerService } from '@core/services/spinner.service';
...
+            .pipe(takeUntil(this.unsubscribe$))
+            .subscribe((resp) => {
+                const [teamSlots] = resp.teamSlots.filter((el) => el.name === this.teamName);
+
+                this.slots = teamSlots.availabilitySlots;
+                this.spinnerService.hide();
+            });
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/team/team.component.ts]: This is not a good idea to reload all data to update 1 team
We have two ways how we can fix it:
1. Load and update all data in the user and teams page (like this idea more)
2. Add new endpoint to load specific team slots

File: frontend/src/app/modules/availability/user-slot/user-slot.component.ts [hunk 6]
```diff
@@ -15,17 +15,15 @@ export class UserSlotComponent {
 
     @Input() public currentUser: IUser;
 
-    protected destroyed$ = new Subject<void>();
-
-    readonly untilThis = <T>(source: Observable<T>) => source.pipe(takeUntil(this.destroyed$));
+    private unsubscribe$ = new Subject<void>();
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/user-slot/user-slot.component.ts]: see the previous comment

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/availability/team/team.component.ts (hunk 5 above)
Comment (Pavliuk13) [reply frontend/src/app/modules/availability/team/team.component.ts]: if I create 2 new endpoints (only for user's slots and only for team's slots), will it be good?

Comment (tatianahutii) [reply frontend/src/app/modules/availability/team/team.component.ts]: You can, but I believe it will be easier to just reload all data
And logic will be simpler

Comment (Pavliuk13) [reply frontend/src/app/modules/availability/team/team.component.ts]: I made it. Now all data is reloaded together

File: frontend/src/app/modules/availability/availability-page/availability-page.component.ts [hunk 7]
```diff
@@ -55,4 +55,10 @@ export class AvailabilityPageComponent {
                 }
             });
     }
+
+    reloadData(isReload: any) {
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/availability-page/availability-page.component.ts]: Please do not use any

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Comment by tatianahutii** [comment]
Could you please delete this from all components you worked at?
You need to use BaseComponent instead of this

image

**Comment by Pavliuk13** [comment]
> Could you please delete this from all components you worked at? You need to use BaseComponent instead of this
> 
> image

I removed it from AvailabilityPageComponent and SlotComponent

File: frontend/src/app/modules/availability/availability-page/availability-page.component.ts (hunk 7 above)
Comment (Pavliuk13) [reply frontend/src/app/modules/availability/availability-page/availability-page.component.ts]: fixed

**Review by tatianahutii — APPROVED** [review]
LGTM
